client/src/components/landing/LandingNavigationBar.vue

```diff
@@ -4,19 +4,25 @@
     <div class="mj-landing-navbar-buttons">
       <language-change-button :is-transparent="true" />
       <login-button />
+      <span class="mj-register-visible">
+        <register-button />
+      </span>
```

Comment:
I would remove the `span` tag - unnecessary. You later on set span display property as `flex`. Span is used if you want to stay in `inline` display property, otherwise, div are better option. In this case to be hones, you can add class directly on component. It will be added to the firs html tag the vue renders when extracting component (in our case - a div).

Plus I don't like the class name for this one. `-visible` suggests that the class is added via js in some point.

Fun fact. For such easy styles, you could use classes `d-none` and `d-sm-block` from bootstrap. More on this in the docs: https://getbootstrap.com/docs/5.1/utilities/display/
```suggestion
        <register-button class="mj-landing-nav-register-button" />
```